test/run-windows-k8s-integration.sh

```diff
@@ -22,7 +21,7 @@ readonly teardown_driver=${GCE_CLUSTER_ZONE:-true}
 make -C ${PKGDIR} test-k8s-integration
 
 base_cmd="${PKGDIR}/bin/k8s-integration-test \
-            --platform=windows --local-k8s-dir=${LOCAL_K8S_DIR} --bringup-cluster=false --teardown-cluster=false --teardown-driver=${teardown_driver}\
+            --platform=windows --local-k8s-dir=${k8s_dir} --bringup-cluster=false --teardown-cluster=false --teardown-driver=${teardown_driver}\
```

Comment:
local k8s dir is only for if you're running the test from your own local machine. Do we need it for CI? k8s-integration should download kubernetes and build in a tmp directory.

Reply:
in prow test using kubetest, kuberentes will be installed under current dir which is the pd csi driver repo. in the test, kubetest needs to be running in kubernetes dir. We don't need to download kuberentes locally.

Reply:
kubetest should be run from `testDir`, not `k8sDir`: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/test/k8s-integration/main.go#L221

Maybe we need to set `--test-version` here.

Btw, does kubernetes_e2e support gke clusters?

Reply:
I see there are some confusion here. For current pd driver ci tests for linux, we build kubernetes and install it under a tmp dir before running the test. But this step is not needed for the windows test setup. It uses script /workspace/scenarios/kubernetes_e2e.py which will install kuberentes like kube_up, I think. https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/gcp-compute-persistent-disk-csi-driver-windows.yaml#L65

This script will install kuberentes under current dir which is pd csi driver repo. The local-k8s-dir setting here does not mean the test will install kubernetes locally. I used this setting because the test use it change current dir here https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/test/k8s-integration/main.go#L421

So because I didn't set kubeversion or testversion, so the test code will not install kubernetes locally because it is not needed. https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/1c51cca720acaa91f2f00e55a5e44d5d5832ed0b/test/k8s-integration/main.go#L227

I plan to have another PR to edit the config parameter so that it is easier to read. Also I think for linux test, we can also use /workspace/scenarios/kubernetes_e2e.py to setup kubernetes instead of building it locally. We can set to use ci/latest or ci/master depending which kubernetes version we want to test.

Reply:
I think we should set "--test-version" instead. The reason is because we may use different Kubernetes versions to 1) bring up the cluster and 2) run e2e binary

We generally use "master" as the e2e version so that we can get new test cases and test fixes without relying on them to be backported to older Kubernetes versions.

"--local-k8s-dir" is intended for developer testing where you just want to run the e2es quickly without having to download and build kubernetes.

Reply:
my understanding is since we already have prebuild kubernetes in different versions https://github.com/kubernetes/test-infra/blob/a4b7e151278a26fb45ee186b7fdc96394deed016/docs/kubernetes-versions.md
we could just use it by setting extract=ci/k8s-beta , ci/k8s-master , ... etc.
It will save k8s build time. Is there some specific version we want to test?

Reply:
Yes, I think in release versions we can save a lot of time by using extract.  But if we want to test head, we still need to build. We do have https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/test/k8s-integration/main.go#L40 that I think can help with this.

But still, we have 3 distinct variables we use to distinguish different use cases:
- kubeVersion to set the cluster version
- testVersion to set the e2e version
- local for local development (you're making a change in kubernetes or the e2es that you want to test)

Reply:
synced up offline. Will remove the need to locally download and build e2e binary in a separate PR. Instead we could set extract=ci/k8s-version.